bot/handlers/role_selection.go

```diff
@@ -40,46 +40,40 @@ func NewRoleSelectionHandler(guildID string, config *config.RoleSelectionConfig)
 }
 
 // ReconcileConfig reconciles the role selection configuration with the Discord guild.
-// TODO: This function is not fully implemented and needs to be more reliable.
 func (r *RoleSelectionHandler) ReconcileConfig(s *discordgo.Session) error {
-	r.reconciling.RLock()
-	chList, err := s.GuildChannels(r.guildID)
+	r.reconciling.Lock()
+	defer r.reconciling.Unlock()
+
+	roleChannel, err := s.Channel(r.config.Channel.ID)
 	if err != nil {
-		r.logger.Error("error getting channels for guild", zap.String("guildID", r.guildID), zap.Error(err))
-		r.reconciling.RUnlock()
+		r.logger.Error("error getting channel", zap.String("channelID", r.config.Channel.ID), zap.Error(err))
 		return err
 	}
-	roleChExists := false
```

Comment:
@jules You probably want some sort of logic to confirm that the role channel exists and to create it or log with an error if it does not exist